### ISSUE TITLE
Append/Get does not work as expected

### ISSUE THREAD

**Issue by pdhammond42** [tebeka/deque#1]
HI, 
Get fails when more than one block is in use. This test case fails:

````
func TestAppendAndGet(t *testing.T) {
	n := 128
	q := deque.New()

	for i := 0; i < n; i++ {
		q.Append(i + 1)
	}

	for i := 0; i < n; i++ {
		o, err := q.Get(i)
		assert.NoError(t, err)
		assert.Equal(t, i+1, o)
	}
}

````

=== RUN   TestInsertAndGet
--- FAIL: TestInsertAndGet (0.00s)
    queue_test.go:93: 
        	Error Trace:	queue_test.go:93
        	Error:      	Not equal: 
        	            	expected: int(2)
        	            	actual  : <nil>(<nil>)
        	Test:       	TestInsertAndGet
...
    queue_test.go:93: 
        	Error Trace:	queue_test.go:93
        	Error:      	Not equal: 
        	            	expected: int(32)
        	            	actual  : <nil>(<nil>)
        	Test:       	TestAppendAndGet
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x12d79db]

**Comment by tebeka** [comment]
Thanks, I'll have a look soon-ish.

I've changed the test a bit to work without `testify` and I see a different output (still a bug)

```go
func TestAppendAndGet(t *testing.T) {
	n := 128
	q := New()

	for i := 0; i < n; i++ {
		q.Append(i + 1)
	}

	for i := 0; i < n; i++ {
		o, err := q.Get(i)
		if err != nil {
			t.Fatal(err)
		}
		if o != i+1 {
			t.Fatalf("get - %d: %v != %v", i, i+1, o)
		}
	}
}
```

Output:
```
=== RUN   TestAppendAndGet
--- FAIL: TestAppendAndGet (0.00s)
    deque_test.go:245: get - 1: 2 != <nil>
```

I *guess* the panic is from `testify`.

**Comment by tebeka** [comment]
Fixed in 56f87eca7c35fcad8549535b7078005713ec6ce9
